### PR TITLE
Make alias protocol_compiler visible

### LINKED ISSUE #1189: [Bug]: gRPC v1.48.1.bcr.2's new `//third_party:protocol_compiler` alias is not visible for `cc_grpc_library` macro
### What happened?

I'm trying to update my code [on this branch](https://github.com/rendezqueue/rendezqueue/tree/grpc-v1.48.1) to use Bazel version 6, which gRPC has been incompatible with for a while. But version `1.48.1.bcr.2` is almost enough! The only trouble is that its `//third_party:protocol_compiler` alias isn't publicly visible and causes errors when using the `cc_grpc_library` macro:

```
ERROR: /home/runner/work/rendezqueue/rendezqueue/proto/BUILD.bazel:16:16: in _generate_cc rule //proto:_rendezqueue_cc_grpc_grpc_codegen: alias '@grpc~1.48.1.bcr.2//third_party:protocol_compiler' referring to target '@protobuf~3.19.6//:protoc' is not visible from target '//proto:_rendezqueue_cc_grpc_grpc_codegen'. Check the visibility declaration of the former target if you think the dependency is legitimate
ERROR: /home/runner/work/rendezqueue/rendezqueue/proto/BUILD.bazel:16:16: Analysis of target '//proto:_rendezqueue_cc_grpc_grpc_codegen' failed
```

Is this intentional? Any objections to making a version `1.48.1.bcr.3` with more visibility?

### Version

Development (host) and target OS/architectures: `Linux [redacted] 5.10.0-25-amd64 #1 SMP Debian 5.10.191-1 (2023-08-16) x86_64 GNU/Linux`

Output of `bazel --version`: bazel 6.4.0

Version of relevant rules from the `WORKSPACE` or `MODULE.bazel` file:  grpc 1.48.1.bcr.2

Language(s) and/or frameworks involved: C++

### How to reproduce

```shell
git clone -b grpc-v1.48.1 https://github.com/rendezqueue/rendezqueue.git
cd rendezqueue
bazel build //proto:rendezqueue_rendezqueue_cc_grpc
```

### Any other information?

_No response_

### PR DESCRIPTION
**Pull request by mering**
Make `//third_party:protocol_compiler` target public as required by macros.

Fixes #1189

### CODE REVIEW

**Comment by alexeagle** [comment]
How was this tested?
Since the previous release had a bug that was only visible in downstream usage, it would be nice to know how you verified there isn't yet another bug present in this release.

**Comment by mering** [comment]
@alexeagle I tested this PR with our code base. Of course I cannot guarantee that we are using all code paths. Given that I only searched and replaced the `protocol_compiler` target in the whole code base as compared to the known working version `1.48.1.bcr.1` and made this target public now, I am quite confident that this should work now.

**Comment by grencez** [comment]
> How was this tested? Since the previous release had a bug that was only visible in downstream usage, it would be nice to know how you verified there isn't yet another bug present in this release.

Downstream user here. The release didn't introduce a new bug per se - it fixed enough for my use case that this one last visibility issue became obvious. And I verified this change is good.

I also don't mind waiting for 1.48.1.bcr.2 to bake longer if anyone's unsure.

File: modules/grpc/1.48.1.bcr.3/source.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "url": "https://github.com/grpc/grpc/archive/refs/tags/v1.48.1.tar.gz",
+    "integrity": "sha256-MgNmZl0ZAnzah7I2jAOTkAajfgOIv9EJHI0qlvvJO9g=",
+    "strip_prefix": "grpc-1.48.1",
+    "patches": {
+        "adopt_bzlmod.patch": "sha256-+XSHXK0VjNIzdW+A3t2PYDKhtl/Ij/7hBQBkH42S2eE="
```
Comment (Wyverald) [inline modules/grpc/1.48.1.bcr.3/source.json:6]: to confirm, you merged the two patches of 1.48.1.bcr.2 together, right?

Comment (mering) [reply modules/grpc/1.48.1.bcr.3/source.json:6]: Yes exactly. I merged the two patches and also included the one-line patch to make the `//third_party:protocol_compiler` target public.

**Comment by Wyverald** [comment]
gonna go ahead and merge now; if more fixes are needed we can always push a .bcr.4

**Review by Wyverald — APPROVED** [review]

**Comment by mering** [comment]
@Wyverald any idea why the test is failing?

**Comment by grencez** [comment]
> any idea why the test is failing?

This seems to be https://github.com/bazelbuild/bazel/issues/10472. The openssl system headers on the macOS test system are getting picked up in some cases, causing conflicts with the typedefs in BoringSSL's headers.

```
In file included from external/boringssl~0.0.0-20211025-d4f1ab9/src/crypto/x509v3/v3_skey.c:62:
In file included from /usr/local/include/openssl/err.h:21:
/usr/local/include/openssl/e_os2.h:183:11: warning: 'OPENSSL_EXPORT' macro redefined [-Wmacro-redefined]
#  define OPENSSL_EXPORT extern
          ^
external/boringssl~0.0.0-20211025-d4f1ab9/src/include/openssl/base.h:222:9: note: previous definition is here
#define OPENSSL_EXPORT
```

**Comment by mering** [comment]
@Wyverald friendly ping

**Comment by Wyverald** [comment]
Just for my information, do you know why 1.48.1.bcr.2 didn't have this failure? It might be due to some mac CI machine upgrades we did recently; just wanted to confirm.

**Comment by grencez** [comment]
> Just for my information, do you know why 1.48.1.bcr.2 didn't have this failure? It might be due to some mac CI machine upgrades we did recently; just wanted to confirm.

Yeah a machine update could have triggered this if `/usr/local/include/openssl/` didn't exist before or contained `base.h`. I guess the opposite is true now (openssl system headers exist and `base.h` is not one of them), causing Clang to use some headers from the system and some from boringssl.

It's also possible that the `--sandbox_block_path=/usr/local` workaround was removed (e.g., maybe the build user's .bazelrc was deleted).

**Comment by mering** [comment]
@Wyverald Could you please trigger the tests again?

I tried applying the workaround from #10472.

**Comment by Wyverald** [comment]
the workaround is unfortunately breaking the windows build; you'll need to separate the CI configs for those platforms

**Comment by mering** [comment]
> the workaround is unfortunately breaking the windows build; you'll need to separate the CI configs for those platforms

Well, it does proof that the problem is bazelbuild/bazel#10472 and in turn unrelated to this PR. So can't we just merge the first commit without the workaround as it works as intended and just the CI is broken?

**Comment by mering** [comment]
I modified the workaround to be applied for macOS only.

**Comment by Wyverald** [comment]
Thanks for pulling through! All looks good now.

**Pull request merged**